lib/Support/MemoryUsage.cpp

```diff
@@ -108,10 +108,20 @@ size_t util::GetTotalMallocUsage() {
 
 #elif defined(HAVE_MALLOC_ZONE_STATISTICS)
 
-  // Support memory usage on Darwin.
-  malloc_statistics_t Stats;
-  malloc_zone_statistics(malloc_default_zone(), &Stats);
-  return Stats.size_in_use;
+  // Memory usage on macOS
+
+  malloc_statistics_t stats;
+  malloc_zone_t **zones;
+  unsigned int num_zones;
+
+  malloc_get_all_zones(0, nullptr, (vm_address_t **)&zones, &num_zones);
```

Comment:
Please check the return value for success (`KERN_SUCCESS` or `0`).

Out of curiosity, do you have any information how the first `task_t` parameter is defined?
I didn't find any additional documentation.